pkg/fqdn/helpers.go

```diff
@@ -52,6 +53,9 @@ func injectToCIDRSetRules(rule *api.Rule, cache *DNSCache, reMap *regexpmap.Rege
 	for egressIdx := range rule.Egress {
 		egressRule := &rule.Egress[egressIdx]
 
+		// Build an IP collection to remove all duplicates
+		allIPs := ip.NewCollection()
```

Comment:
Could you make this a `[]net.IP` and use `keepUniqueIPs` (https://github.com/cilium/cilium/blob/master/pkg/ip/ip.go#L730) on line 111, where we finally add toCIDRSet rules?